frontend/src/app/hooks/useCampaignForm.ts

```diff
@@ -0,0 +1,114 @@
+"use client";
+import { useState } from "react";
+import { API_BASE, postForm } from "@/lib/api";
+import { useRouter } from "next/navigation";
+
+export default function useCampaignForm() {
+  const router = useRouter();
+
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [targetAmount, setTargetAmount] = useState<number>(0);
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [status, setStatus] = useState("pending");
+  const [image, setImage] = useState<File | null>(null);
+
+  const [submitting, setSubmitting] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  async function uploadFileToS3(file: File) {
+    try {
+      const response = await fetch(`${API_BASE}/api/campaigns/upload-url`, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ filename: file.name, contentType: file.type }),
+      });
+      if (!response.ok) throw new Error(await response.text());
+
+      const { url, key } = await response.json();
+
+      const put = await fetch(url, {
+        method: "PUT",
+        body: file,
+        headers: { "Content-Type": file.type },
+      });
+
+      if (!put.ok) throw new Error("Failed to upload to S3");
+
+      return key;
+    } catch (err) {
+      console.error(err);
+      throw err;
+    }
+  }
+
+  async function onSubmit(e: React.FormEvent) {
+    e.preventDefault();
+    setError(null);
+
+    if (
+      !title ||
+      !description ||
+      !targetAmount ||
+      !startDate ||
+      !endDate ||
+      !image
+    ) {
+      setError("All fields including image are required");
+      return;
+    }
+
+    const orgId = localStorage.getItem("id"); // get org_id from localStorage
```

Comment:
the ID local storage item is not a descriptve name, rename it to ORG_ID or shomthing more descriptive,